test/test.js

```diff
@@ -357,6 +358,85 @@ test('--output with invalid path fails', async t => {
   }
 });
 
+test('--junit with empty input has single successful test', async t => {
+  const input = '';
+  const junit = '../junitA.xml';
+  const result = await execa('../markdownlint.js',
+    ['--stdin', '--junit', junit],
+    {input, stripFinalNewline: false});
+  const xml = fs.readFileSync(junit, 'utf8');
+  const parsedXml = xmlParser.xml2js(xml, {compact: true});
+  t.is(result.stdout, '');
+  t.is(result.stderr, '');
+  t.is(Object.keys(parsedXml.testsuites).length, 1);
```

Comment:
See my separate comment about maybe just comparing the output text directly. What you have here is more correct and thorough, but it’s a lot more typing and harder to follow. I’m fine with either choice - up to you if you want to type more. :)